views/forms/ChartFilterForm/index.jsx

```diff
@@ -1,83 +1,81 @@
-import React, { useState } from 'react'
+import React, { useEffect } from 'react'
 
-import { Button, FormControl, InputLabel } from '@mui/material'
+import { FormControl, InputLabel } from '@mui/material'
 import { useForm } from 'react-hook-form'
 
-import { SITES_BY_NETWORK } from '../../../constants'
-import { FILTER_CATEGORIES } from '../../../constants/vars'
+import {
+  SITES_BY_NETWORK,
+  NETWORKS_FILTER_KEY,
+  SITES_FILTER_KEY,
+  FILTER_CATEGORIES,
+} from '../../../constants'
+import { ChartFiltersModel } from '../../models'
 import DropdownCheckboxGroup from '../DropdownCheckboxGroup'
-import './ChartFilterForm.css'
 
-const NETWORKS_FILTER_KEY = 'networks'
-const SITES_FILTER_KEY = 'sites'
+import './ChartFilterForm.css'
 
 const ChartFilterForm = ({ initialValues, onSubmit }) => {
-  const { handleSubmit, setValue, getValues } = useForm({
-    defaultValues: initialValues,
+  const { handleSubmit, setValue, watch, control } = useForm({
+    defaultValues: ChartFiltersModel.filtersToFormValues(initialValues),
   })
-  const formValues = getValues()
-  const [resetSiteOptions, setResetSiteOptions] = useState(false)
-
-  const handleChange = (label, value) => {
-    const newFilterValues = Object.keys(initialValues[label]).reduce(
-      (acc, key) => {
-        acc[key] = value.includes(key)
-          ? { label: key, value: 1 }
-          : { label: key, value: 0 }
-        return acc
-      },
-      {}
+  const handleClear = (filterField) => setValue(filterField, [])
+  const handleSelectAll = (filterField) => {
+    const allFilterValues = Object.values(initialValues[filterField]).map(
+      ({ label }) => label
     )
-    if (label === NETWORKS_FILTER_KEY) {
-      const allowedSiteOptions = { ...initialValues.sites }
-      const allowedSites = value.flatMap((network) => SITES_BY_NETWORK[network])
-      const selectableSites = Object.keys(initialValues.sites)
-      for (const site of selectableSites) {
-        if (allowedSites.length && !allowedSites.includes(site)) {
-          delete allowedSiteOptions[site]
-        }
-      }
-      setValue(SITES_FILTER_KEY, allowedSiteOptions)
-      setResetSiteOptions(true)
-    }
-    setValue(label, newFilterValues)
+    return setValue(filterField, allFilterValues)
   }
 
+  useEffect(() => {
+    const subscription = watch((value, { name }) => {
+      if (name === NETWORKS_FILTER_KEY) {
+        const sitesUnion = value[NETWORKS_FILTER_KEY].reduce(
+          (siteList, network) => {
+            return siteList.concat(SITES_BY_NETWORK[network])
```

Comment:
are there any sites that exist across multiple networks?

Reply:
That's a good catch.

[1672b0a](https://github.com/AMP-SCZ/dpdash/pull/738/commits/1672b0adc912596b356011dcd0644aa172c75b0a) 

should take care of it.